.github/workflows/code-analysis.yml

```
@@ -0,0 +1,31 @@
+name: PHP Composer
+
+on:
+  pull_request:
+    branches: [ main ]
+
+jobs:
+  php-stan:
+    runs-on: ubuntu-latest
+    steps:
+    - name: Install packages and run PHP Stan
+      uses: actions/checkout@v2
+    - run: composer install
+    - run: sudo cp vendor/prestashop/php-dev-tools/phpstan/ps-module-extension.neon vendor/prestashop/php-dev-tools/phpstan/extension.neon
+    - run: sudo cp vendor/prestashop/php-dev-tools/templates/phpstan/phpstan.neon phpstan.neon
+    - run: sudo chown -R $USER:$USER phpstan.neon
+    - run: sudo echo '  excludes_analyse:' >> phpstan.neon
+    - run: sudo echo '    - %currentWorkingDirectory%/vendor/*' >> phpstan.neon
+    - run: _PS_ROOT_DIR_=./ php vendor/phpstan/phpstan/phpstan.phar --configuration=phpstan.neon analyse ./
+  
+  php-cs-fixer:
+    name: PHP-CS-Fixer
+    runs-on: ubuntu-latest
+    steps:
+      - name: Install packages and run PHP CS Fixer
+        uses: actions/checkout@v2.0.0
+      - run: composer install
+      - run: sudo cp vendor/prestashop/php-dev-tools/templates/cs-fixer/php_cs.dist .php_cs.dist
+      - run: sudo cp vendor/prestashop/php-dev-tools/templates/cs-fixer/prettyci.composer.json .prettyci.composer.json
+      - run: php vendor/bin/php-cs-fixer fix
```

GitHub Action should only check if the style has been correctly applied:
```suggestion
      - run: php vendor/bin/php-cs-fixer fix --dry-run
```